src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -142,5 +145,59 @@ public override bool CanApplyChange(ApplyChangesKind feature)
         {
             return true;
         }
+
+        protected override void ApplyDocumentRemoved(DocumentId documentId)
+        {
+            var document = this.CurrentSolution.GetDocument(documentId);
+            if (document != null)
+            {
+                DeleteDocumentFile(document.Id, document.FilePath);
+                this.OnDocumentRemoved(documentId);
+            }
+        }
+
+        private void DeleteDocumentFile(DocumentId id, string filePath)
+        {
+            try
+            {
+                File.Delete(filePath);
+            }
+            catch (IOException) { }
+            catch (NotSupportedException) { }
+            catch (UnauthorizedAccessException) { }
```

Comment:
Do we want to log any of these for debugging purposes?

Reply:
Good point, I'll log these.